SeaPkg/Library/BasePeCoffLibNegative/BasePeCoffLibNegative.c

```diff
@@ -880,8 +881,21 @@ PeCoffImageDiffValidation (
         }
 
         // Inspection of the memory attributes of the target image
+        if (ImageValidationEntryHdr->Size > MAX_ENTRY_SIZE) {
+          DEBUG ((
+            DEBUG_ERROR,
+            "%a: A new cfg entry is larger than the current max size!  MAX_ENTRY_SIZE: %x  Entry Size: %x\n",
+            __func__,
+            MAX_ENTRY_SIZE,
+            ImageValidationEntryHdr->Size
+            ));
+            Status = EFI_BUFFER_TOO_SMALL;
+            break;
+        }
         AddrInTarget = 0;
-        CopyMem (&AddrInTarget, (UINT8 *)TargetImage + ImageValidationEntryHdr->Offset, ImageValidationEntryHdr->Size);
+        ZeroMem (Target, MAX_ENTRY_SIZE);
+        CopyMem ((VOID *)&Target[0], (UINT8 *)TargetImage + ImageValidationEntryHdr->Offset, ImageValidationEntryHdr->Size);
+        AddrInTarget = (EFI_PHYSICAL_ADDRESS)*Target;

```

Comment:
It seems that we only care about the first 8 bytes here? If so, why do we need to set a stellar upper limit (80) for the buffer size?